enhancements/network/ovn-secondary-networks.md

```diff
@@ -0,0 +1,240 @@
+---
+title: ovn-secondary-networks
+authors:
+  - "@maiqueb"
+reviewers:
+  - "@dcbw"
+  - "@dougbtv"
+  - "@squeed"
+  - "@trozet"
+  - "@abhat"
+  - "@alexanderConstantinescu"
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-06-15
+last-updated: yyyy-mm-dd
+status: provisional
+---
+
+# OVN secondary networks
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+This enhacement proposal studies and describes how to support OVN as the
+networking provider for a Multus secondary interface.
+
+## Motivation
+The networking plugins that can be used as secondary interfaces are currently
+quite simple, and are unable to provide advanced networking requirements.
+
+Furthermore, virtualization users are used to being able to define isolated
+tenant networks that form a single L2 domain. OVN excels at this, by creating
+simple L2 overlays without the need for a network administrator.
+
+### Goals
+- create flat layer 2 overlays
+- connect a newly created pod to these overlay networks
+- these overlays can optionally be a subnet
+- IPv6 support is welcome (not an immediate priority)
+- optional connection to the external networks
+
+### Non-Goals
+- network policy concerns. There isn't a requirement for having network
+  policies for the secondary networks. There is an
+  [enhancement proposal](enhancements/network/multi-networkpolicy.md) about
+  this subject, not scoped around OVN.
+
+## Proposal
+
+### User Stories
+
+#### Isolated tenant network
+A customer can specify a layer 2 overlay network with a subnet via a
+`network-attachment-definition`, which is provisioned without the intervention
+of a system administrator. You can refer to the
+[network attachment definition section](#net-attach-def-samples) for examples.
+
+Afterwards, the users can then request their pods - and / VMs - to connect to
+those overlays as they would for any other secondary network.
+
+### Implementation Details/Notes/Constraints
+A new CRD will be introduced, named `OVNSecondaryNetwork`. When this CRD is
+provisioned, a `NetworkAttachmentDefinition` object will be rendered.
+
+The new CRD will look like:
+
+```golang
+type OVNSecondaryNetwork struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Spec OVNSecondaryNetworkSpec `json:"spec,omitempty"`
+}
+
+type OVNSecondaryNetworkSpec struct {
+    // Subnet is a RFC 4632/4291-style string that represents an IP address and prefix length in CIDR notation
+    Subnet                  string `json:"subnet"`
+    HasExternalConnectivity bool   `json:"hasExternalConnectivity,omitempty"`
```

Comment:
Should spell out the guarantees around this. Is traffic going to be NAT-ed? Directly dumped onto the host's underlay? If NAT then what IP does it get NAT-ed to? Are there any guarantees around ingress to this network, or is only reply traffic that is allowed?

Reply:
Perhaps ExternalConnectivity should be its own struct, to allow for expansion? What potential options are there?

- no external access
- using the node's gateway router
- A centralized gateway

Reply:
Using the node's gateway router, afaiu, implies that we will be sharing the OVN db between OVN-K and whatever implements these overlays, right ? 

That would only be an option **if** this feature is implemented in OVN-Kubernetes - and, imo, is one of the advantages of implementing it there - i.e. we wouldn't **need** tp implement a centralized gateway to provide external connectivity to the pods.

Never-the-less, I think having `ExternalConnectivity` as a separate struct is a good suggestion, I'll model it; thanks!